### PR TITLE
Fix zef install error holdover from version 0.0.1

### PR DESCRIPTION
**Pull request by tbrowder**
See the Changes file for details.

Also added File::temp as a test dependency.

The fix was to use the is-deeply test on sub-directories of the two top-level collections rather than from the top-level directly.

This should fix issue #9.

### CODE REVIEW

**Comment by lizmat** [comment]
I think the test.yml needs a `zef install --deps-only` ?

**Comment by tbrowder** [comment]
Could be. I didn't check that.

**Comment by tbrowder** [comment]
Well that was weird, but at least it works now. I used to always put all deps in depends, now I hope I remember to do that again!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -24,5 +24,10 @@ jobs:
       - uses: Raku/setup-raku@v1
         with:
           raku-version: ${{ matrix.raku-version }}
+      - name: Install Dependencies
+        run: |
```
Comment (lizmat) [inline .github/workflows/test.yml]: Is this | needed?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,5 +24,10 @@ jobs:
...
+          zef install --/test --deps-only .
       - name: Run Tests
         run: raku run-tests
+      - name: Install Module
```
Comment (lizmat) [inline .github/workflows/test.yml:31]: Why do we test installing?

Comment (tbrowder) [reply .github/workflows/test.yml:31]: I will fix it.

**Comment by tbrowder** [comment]
Lizmat, are you going to use @ugexe's install method instead of your 'raku run-tests'?

**Comment by lizmat** [comment]
`raku run-tests` is intended to be run by a. the author, and b. by the CI logic.

It has nothing to do with installing a module.

**Comment by tbrowder** [comment]
I was just wondering why you haven’t published it yet. Shall I go ahead and create a second PR for the 3-OS test now?

**Comment by lizmat** [comment]
The reason didn't publish "run-tests" is that I don't want to create a dependency.  It comes with the module, for simplicity's sake.  Feel free to copy it :-)

**Comment by lizmat** [comment]
> Shall I go ahead and create a second PR for the 3-OS test now?

Sure  :-)

**Comment by ugexe** [comment]
Everyone is free to test how they want, but as I've been mentioned I'd like to clarify my personal stance.

I would encourage people to *not* run a special test script for CI testing. You *want* to test the install unless you are 100% sure you are testing every code path. I say this because there are probably at least 10 closed issues in zef where a user complains that their tests pass locally but when they try to install it later via `zef install Foo` it ultimately fails during precompilation. These are issues with the authors code, and they were not aware of it because a) their tests are lacking (such as some modules never being used during testing) b) they never tried to install it (which would have ensured every file was precompiled). At this point it doesn't take me that much time to triage when users post issues in the zef repo, but over the years I've spent multiple hours debugging this issue over and over.

**Comment by lizmat** [comment]
I'll add a `zef install . --/test` to the "run-tests" script, to be activated by an "--install" argument that will be set in the test.yml

The reason I want to keep using my own script in there:
a. no dependency, so faster testing and turnaround, less CPU usage for testing, especially for modules that do not have any dependencies of their own
b. clear which version of Rakudo / MoarVM is being used to test
c. if there are tests failing, I get the full output of the failed test-file, including any `--ll-exception` stacktrace, but **not** any output from any test-files that did not fail.  So less noise.

**Comment by ugexe** [comment]
> no dependency, so faster testing and turnaround, less CPU usage for testing, especially for modules that do not have any dependencies of their own

I think zef is already installed in the base github actions image. If so then I'm pretty sure using `zef install .` (potentially using `--/update` if there are no dependencies it would need to search for) would both use less resources and less cpu than using `run-tests` and a separate install step. There is something to be said about zef not running xt/ tests, but I think `prove6` is also installed already which I imagine has some combination of flags to allow less verbose output. It can make sense to have a custom test script for local testing, but an optimal CI testing setup would not do that. However, I can recognize it can sometimes be convenient to use the same workflow for local and CI testing

**Pull request merged**